force-app/main/default/lwc/lmsPublisherWebComponent/__tests__/lmsPublisherWebComponent.test.js

```diff
@@ -0,0 +1,129 @@
+                    'c-contact-list-item-bubbling'
+                );
+                expect(detailEls.length).toBe(mockGetContactList.length);
+            });
+        });
+
+        it('renders with no record', () => {
+            // Create initial element
+            const element = createElement('c-lms-publisher-web-component', {
+                is: LmsPublisherWebComponent
+            });
+            document.body.appendChild(element);
+
+            // Emit data from @wire
+            getContactListAdapter.emit(mockGetContactListNoRecords);
+            // Return a promise to wait for any asynchronous DOM updates. Jest
+            // will automatically wait for the Promise chain to complete before
+            // ending the test and fail the test if the promise rejects.
+            return Promise.resolve().then(() => {
+                // Select elements for validation
+                const detailEls = element.shadowRoot.querySelectorAll(
+                    'c-contact-list-item-bubbling'
+                );
+                expect(detailEls.length).toBe(
+                    mockGetContactListNoRecords.length
+                );
+            });
+        });
+    });
+
+    it('calls publish based on an event from a child c-contact-list-item-bubbling component', () => {
+        const CONTACT = {
+            Id: '0031700000pJRRSAA4',
+            Name: 'Amy Taylor',
+            Title: 'VP of Engineering',
+            Phone: '4152568563',
+            Email: 'amy@demo.net',
+            Picture__c:
+                'https://s3-us-west-1.amazonaws.com/sfdc-demo/people/amy_taylor.jpg'
+        };
+
+        // Message Channel used in component
+        const MESSAGE_CHANNEL = recordSelected;
+
+        const PAYLOAD = { recordId: CONTACT.Id };
+
+        // Create initial element
+        const element = createElement('c-lms-publisher-web-component', {
+            is: LmsPublisherWebComponent
+        });
+        document.body.appendChild(element);
+
+        // Emit data from @wire
+        getContactListAdapter.emit(mockGetContactList);
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+        return Promise.resolve().then(() => {
+            const detailEl = element.shadowRoot.querySelector(
+                'c-contact-list-item-bubbling'
+            );
+            // Dispatch new event on child component to validate if it triggers
+            // a publish call in the current component.
+            detailEl.dispatchEvent(
+                new CustomEvent('contactselect', {
+                    detail: CONTACT,
+                    bubbles: true
+                })
+            );
+
+            // Was publish called and was it called with the correct params?
+            expect(publish).toHaveBeenCalled();
```

Comment:
Duplicate check, use the second one only.

Reply:
Done.